src/components/CategorizedTodos.js

```diff
@@ -0,0 +1,61 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import Todo from './Todo';
+
+export default function CategorizedTodos({ todos, setTodos, setEditItem }) {
+  const [categorizedTodos, setCategorizedTodos] = useState({});
+
+  const categoryGroups = () => {
+    // const sortedObj = {};
+    // array.forEach((todo) => {
+    //   // LOOKING to see if a key of item exists
+    //   if (todo.category in sortedObj) {
+    //   // if so, set the value of the key array to the spreaded object and the current todo
+    //     sortedObj[todo.category] = [...sortedObj[todo.category], todo];
+    //   } else {
+    //   // if not, create the category and set the value as the current todo
+    //     sortedObj[todo.category] = [todo];
+    //   }
+    // });
+
+    // DOING THE SAME AS ABOVE USING REDUCE
+    const sortedObj = todos.reduce((todoObject, currentObject) => {
+      const main = todoObject;
+      // if the current category already exists, push the currentObject into the array...otherwise, set the value to an array and push the currentObject into it.
+      (main[currentObject.category] = main[currentObject.category] || []).push(
+        currentObject,
+      );
+      return main;
+    }, {});
+
+    setCategorizedTodos(sortedObj);
+  };
+
+  useEffect(() => {
+    categoryGroups();
+  }, [todos]);
+
+  return (
+    <div>
+      {Object.keys(categorizedTodos).map((category) => (
```

Comment:
You could also use `Object.entries()` It gives you access to the key and value in an array so you'd have to destructure like this.

```js
Object.keys(categorizedTodos).map(([key, value]) => (
```

So on [line 43](https://github.com/nss-evening-cohort-16/YOU-DO-react/pull/13/files#diff-0ff22575e7890bbe18928d04af22980af74f8b6dee8a92d86dcaa8f37391c7d6R43) you don't have to grab the value using bracket notation. You'd instead do this

```js
{value.map((todo) => (
```

Reply:
I just looked it up, and the way you did it was more performant 🙃